### PR TITLE
Add clang-14 for installing gems with Rust extensions

### PR DESCRIPTION
**Pull request by HarlemSquirrel**
Add clang-14 dependency for installing gems with Rust extensions such as https://github.com/HarlemSquirrel/tzf-rb

### CODE REVIEW

**Comment by JalexChen** [comment]
Hi @HarlemSquirrel,

I've ran this and it looks like it's passing. It looks like the link you're referencing is a personal project. Are there other extensions that would need to include this? Would gcc suffice or is clang a requirement here?

**Comment by HarlemSquirrel** [comment]
> Hi @HarlemSquirrel,
> 
> I've ran this and it looks like it's passing. It looks like the link you're referencing is a personal project. Are there other extensions that would need to include this? Would gcc suffice or is clang a requirement here?

I believe anything using https://github.com/oxidize-rb/rb-sys/ would need this. That's the defacto way to build Ruby gems with Rust extensions. This is now a supported feature of RubyGems as of https://github.com/rubygems/rubygems/pull/5175

**Comment by HarlemSquirrel** [comment]
@JalexChen It seems `gcc` is already installed on this image and does not seem to be sufficient.

File: 3.2/Dockerfile [hunk 1]
```diff
@@ -50,6 +50,8 @@ RUN sudo apt-get update && sudo apt-get install -y --no-install-recommends \
 		zlib1g-dev \
 		# YJIT dep
 		rustc \
+		# Build gems with Rust extensions dep
+		clang-14 \
```
Comment (JalexChen) [inline 3.2/Dockerfile:54]: Could you please remove this from the 3.2/Dockerfile and instead place these lines in `Dockerfile.template`?

The next update to ruby will include these updates. After the change, you can run `./shared/gen-dockerfiles.sh 3.2.2` for a Dockerfile you can use

**Review by JalexChen — CHANGES_REQUESTED** [review]

**Review by JalexChen — APPROVED** [review]

**Pull request merged**